client.js

```diff
@@ -131,19 +137,45 @@ function DHT (opts) {
   self._rotateInterval.unref && self._rotateInterval.unref()
 
   process.nextTick(function () {
+    var persistedNodes = []
+    if (opts.persist !== false) {
+      try {
+        persistedNodes = require(self.persistFile)
+        if (!Array.isArray(persistedNodes)) {
+          persistedNodes = []
+          debug('error reading ' + self.persistFile + ': returned object is not an array')
+        }
```

Comment:
This `if` block should be moved out of the `try-catch`. The only part we *expect* to throw an exception is the call to `require`, so everything else should live outside the `try-catch`.